symengine/symengine_exception.h

```diff
@@ -0,0 +1,34 @@
+#ifndef SYMENGINE_EXCEPTION_H
+#define SYMENGINE_EXCEPTION_H
+
+enum symengine_exceptions_t {
+    NO_EXCEPTION,
+    DIV_BY_ZERO,
+    NOT_IMPLEMENTED,
+    UNDEFINED
+};
+
+namespace SymEngine
+{
+
+class SymEngineException : public std::exception
+{
+    std::string m_msg;
+    symengine_exceptions_t ec;
+
+public:
+    SymEngineException(const std::string &msg, symengine_exceptions_t error)
```

Comment:
@rajithv yes, that's exactly what I had in mind. Thanks for doing it, I think it looks very good now.